### DISCUSSION TITLE
Default filter

### DISCUSSION THREAD

**Discussion by siebsie23** [rappasoft/laravel-livewire-tables#857] [Q&A]
Hello,
It seems like setting a default filter is not possible anymore in v2.x.
Is there any workaround for this or will it be added in the future?

EDIT:
I found the new way of applying a filter. For anyone else wondering how to do this: [setFilter()](https://rappasoft.com/docs/laravel-livewire-tables/v2/datatable/events#content-setfilter)

**Comment by dgillier** [comment]
Hello, 
Thanks, where are you sending the event ?
It's not working in the configure function...

**Reply by siebsie23** [reply]
I discovered the same issue. Right now I'm using a workaround that redirects the user to a url with parameters if there are none set.

`if(!$_GET) {
          return redirect('tickets?table[filters][status]=1');
}
return view('tickets::index');`

**Reply by dgillier** [reply]
Clever :)
Thanks, working !

I just hope the setfilter problem will be fixed...

**Comment by cytech** [comment]
this worked for me by calling the setFilter helper. in the builder instance
`
public function builder(): Builder
{
    $this->setFilter('status'), $this->reqstatus);
    return $this->Model::query()
    ....
}
`

**Reply by siebsie23** [reply]
This does work, but when setting multiple filters it gets buggy.

Removing 1 and then removing the other filter will re enable the first one you disabled. Only clicking the 'Clear' button will remove all the filters.

**Comment by cytech** [comment]
indeed it does...
thanx

**Comment by tonypartridge** [comment]
I have done it with the following method, it's far from ideal as it causes a quick redirect by the looks of it:
```php
    public function mount() {
        if(!$this->getAppliedFilterWithValue('created_at')) {
            $this->setFilter('created_at', '2022-10-01');
        }
    }
```
